### PR TITLE
UI: simplify button prompts

### PR DESCRIPTION
**Pull request by illusionyy**

### CODE REVIEW

**Comment by illusionyy** [comment]
Making draft because haven't test

**Comment by illusionyy** [comment]
Tested, seems good

**Comment by bucanero** [comment]
related to this, I remember someone asking to detect the current system settings (button X/O to confirm)

perhaps we can take a look at it... detect the "OK" button and then display the help and handle as needed

(On PS3 and Vita I know you can get the button info, on ps4 I don't remember so we need to check the APIs)

**Comment by illusionyy** [comment]
I can add it to this and Apollo PR. I have asia region console.

**Comment by illusionyy** [comment]
image
image

**Comment by illusionyy** [comment]
@bucanero implemented.

File: source/exec_cmd.c [hunk 1]
```diff
@@ -196,12 +197,28 @@ static void updLocalPlugins(const char* upd_path)
 	}
 }
 
+static struct tm get_local_time(void)
+{
+	static int tz_offset = 0xDEADBEEF;
+
+	if (tz_offset == 0xDEADBEEF)
```
Comment (illusionyy) [inline source/exec_cmd.c]: If this is global variable it would make sense to check it but the variable is initialized above it.

File: source/exec_cmd.c [hunk 2]
```diff
@@ -196,12 +197,28 @@ static void updLocalPlugins(const char* upd_path)
...
+	{
+		int ret = sceSystemServiceParamGetInt(ORBIS_SYSTEM_SERVICE_PARAM_ID_TIME_ZONE, &tz_offset);
+		if (ret < 0)
+		{
+			LOG("Failed to obtain ORBIS_SYSTEM_SERVICE_PARAM_ID_TIME_ZONE! Setting timezone offset to 0");
+			tz_offset = 0;
+		}
+	}
+
+	time_t modifiedTime = time(NULL) + (tz_offset * 60);
```
Comment (illusionyy) [inline source/exec_cmd.c]: For the record, this currently does not handle daylight time savings. Time in US for instance is still 1 hour off but is good enough internationally

Comment (illusionyy) [reply source/exec_cmd.c]: I will have to check if there is a way to get if daylight time savings is active or not. time_t has it but not sure if the system reports it that way

File: source/exec_cmd.c (hunk 1 above)
Comment (bucanero) [reply source/exec_cmd.c]: Actually, it's a local `static` variable so it will hold `0xDEADBEEF` only during the first call to the function.

During that first call, it will enter the `if(...) { }` and we call SystemService to get the actual `tz_offset` value, or set it to `0` if failed. But every future call to `get_local_time ()` will have `tz_offset` set to `0` or the offset value, but no `0xdeadbeef` anymore.

Comment (illusionyy) [reply source/exec_cmd.c]: `static` in this case mean the variable can be cached? Today I learned

Comment (bucanero) [reply source/exec_cmd.c]: > `static` in this case mean the variable can be cached? Today I learned

Yes, it behaves pretty much like a global variable, but the visibility is restricted to the function.

btw, if you want check some notes here:
https://www.tutorialspoint.com/what-are-the-local-static-variables-in-c-language#

**Comment by illusionyy** [comment]
Tested, seems good. also fixed the settings page
image

File: source/exec_cmd.c (hunk 2 above)
Comment (illusionyy) [reply source/exec_cmd.c]: Seems to still report not DST. ah well
```c
show_message("t.tm_year: %d\nt.tm_mon: %d\nt.tm_mday: %d\nt.tm_hour: %d\nt.tm_min: %d\nt.tm_sec: %d\nt.tm_isdst: %d", t.tm_year+1900, t.tm_mon+1, t.tm_mday, t.tm_hour, t.tm_min, t.tm_sec, t.tm_isdst);
```
image
image

Comment (illusionyy) [reply source/exec_cmd.c]: nevermind
```c
	int32_t dst = 0;
	sceSystemServiceParamGetInt(ORBIS_SYSTEM_SERVICE_PARAM_ID_SUMMERTIME, &dst);
	show_message("t.tm_year: %d\nt.tm_mon: %d\nt.tm_mday: %d\nt.tm_hour: %d\nt.tm_min: %d\nt.tm_sec: %d\nt.tm_isdst: %d\nORBIS_SYSTEM_SERVICE_PARAM_ID_SUMMERTIME: %d", t.tm_year+1900, t.tm_mon+1, t.tm_mday, t.tm_hour, t.tm_min, t.tm_sec, t.tm_isdst, dst);
```
image
image
image
image

Comment (illusionyy) [reply source/exec_cmd.c]: Removed check because if user changes timezone settings after first backup, settings won't be updated until user restarts app

Comment (bucanero) [reply source/exec_cmd.c]: > Removed check because if user changes timezone settings after first backup, settings won't be updated until user restarts app

well that's a very rare edge case, might never happen unless you're doing tests. 😄  
(I guess that date/time setup is something that the user will only be doing once when installing the system)

File: source/exec_cmd.c [hunk 3]
```diff
@@ -196,14 +197,36 @@ static void updLocalPlugins(const char* upd_path)
...
+	int ret_dst = sceSystemServiceParamGetInt(ORBIS_SYSTEM_SERVICE_PARAM_ID_SUMMERTIME, &tz_dst);
+	if (ret_dst < 0)
+	{
+		LOG("Failed to obtain ORBIS_SYSTEM_SERVICE_PARAM_ID_SUMMERTIME! Setting timezone daylight time savings to 0");
+		LOG("sceSystemServiceParamGetInt: 0x%08X", ret_dst);
+		tz_dst = 0;
+	}
+
+	time_t modifiedTime = time(NULL) + ((tz_offset + (tz_dst * 60)) * 60);
+	return (*gmtime(&modifiedTime));
```
Comment (bucanero) [inline source/exec_cmd.c]:
```suggestion
	int tz_offset = 0;
	int tz_dst = 0;
	int ret;

	if ((ret = sceSystemServiceParamGetInt(ORBIS_SYSTEM_SERVICE_PARAM_ID_TIME_ZONE, &tz_offset)) < 0)
	{
		LOG("Failed to obtain ORBIS_SYSTEM_SERVICE_PARAM_ID_TIME_ZONE! Setting timezone offset to 0");
		LOG("sceSystemServiceParamGetInt: 0x%08X", ret_tz);
		tz_offset = 0;
	}

	if ((ret = sceSystemServiceParamGetInt(ORBIS_SYSTEM_SERVICE_PARAM_ID_SUMMERTIME, &tz_dst)) < 0)
	{
		LOG("Failed to obtain ORBIS_SYSTEM_SERVICE_PARAM_ID_SUMMERTIME! Setting timezone daylight time savings to 0");
		LOG("sceSystemServiceParamGetInt: 0x%08X", ret_dst);
		tz_dst = 0;
	}

	time_t modifiedTime = time(NULL) + ((tz_offset + (tz_dst * 60)) * 60);
	return (*gmtime(&modifiedTime));
```

File: source/exec_cmd.c [hunk 4]
```diff
@@ -196,14 +197,37 @@ static void updLocalPlugins(const char* upd_path)
...
+static struct tm get_local_time(void)
+{
+	int tz_offset = 0;
+	int tz_dst = 0;
+	int ret;
+
+	if ((ret = sceSystemServiceParamGetInt(ORBIS_SYSTEM_SERVICE_PARAM_ID_TIME_ZONE, &tz_offset)) < 0)
+	{
+		LOG("Failed to obtain ORBIS_SYSTEM_SERVICE_PARAM_ID_TIME_ZONE! Setting timezone offset to 0");
+		LOG("sceSystemServiceParamGetInt: 0x%08X", ret_tz);
```
Comment (bucanero) [inline source/exec_cmd.c]:
```suggestion
		LOG("sceSystemServiceParamGetInt: 0x%08X", ret);
```

File: source/exec_cmd.c [hunk 5]
```diff
@@ -196,14 +197,37 @@ static void updLocalPlugins(const char* upd_path)
...
+		tz_offset = 0;
+	}
+
+	if ((ret = sceSystemServiceParamGetInt(ORBIS_SYSTEM_SERVICE_PARAM_ID_SUMMERTIME, &tz_dst)) < 0)
+	{
+		LOG("Failed to obtain ORBIS_SYSTEM_SERVICE_PARAM_ID_SUMMERTIME! Setting timezone daylight time savings to 0");
+		LOG("sceSystemServiceParamGetInt: 0x%08X", ret_dst);
```
Comment (bucanero) [inline source/exec_cmd.c]:
```suggestion
		LOG("sceSystemServiceParamGetInt: 0x%08X", ret);
```

File: source/menu_options.c [hunk 6]
```diff
@@ -32,7 +33,7 @@ static void _draw_OptionsMenu(u8 alpha)
 				DrawTexture(&menu_textures[c], OPTION_ITEM_OFF - 29, y_off, 0, menu_textures[c].width, menu_textures[c].height, 0xFFFFFF00 | alpha);
 				break;
 			case APP_OPTION_CALL:
-				DrawTexture(&menu_textures[footer_ico_cross_png_index], OPTION_ITEM_OFF - 29, y_off+2, 0, menu_textures[footer_ico_cross_png_index].width/2, menu_textures[footer_ico_cross_png_index].height/2, 0xFFFFFF00 | alpha);
+				DrawTexture(orbisPadGetConf()->crossButtonOK ? &menu_textures[footer_ico_cross_png_index] : &menu_textures[footer_ico_circle_png_index], OPTION_ITEM_OFF - 29, y_off+2, 0, menu_textures[footer_ico_cross_png_index].width/2, menu_textures[footer_ico_cross_png_index].height/2, 0xFFFFFF00 | alpha);
```
Comment (bucanero) [inline source/menu_options.c]:
```suggestion
				DrawTexture(&menu_textures[orbisPadGetConf()->crossButtonOK ? footer_ico_cross_png_index : footer_ico_circle_png_index], OPTION_ITEM_OFF - 29, y_off+2, 0, menu_textures[footer_ico_cross_png_index].width/2, menu_textures[footer_ico_cross_png_index].height/2, 0xFFFFFF00 | alpha);
```

**Review by bucanero — APPROVED** [review]
